### PR TITLE
Prepare release v1.2

### PR DESCRIPTION
**Pull request by rs028**

### CODE REVIEW

**Comment by rs028** [comment]
@spco These `.gitignore` files from the `travis/tests` subdirectory should not be needed and can be removed. However this creates a bunch of empty directories when constraints are not used in a test. Can/should these be removed too?

File: travis/run_tests.sh [hunk 1]
```diff
@@ -208,28 +208,28 @@ done
 if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then bash <(curl -s https://codecov.io/bash) -F tests ; fi
 
 # After all tests are run, exit with a FAIL if $fail_counter>0, otherwise PASS.
-if [[ "$style_test_passed" -gt 0 ]]; then
```
Comment (spco) [inline travis/run_tests.sh:211]: Are these lines meant to be commented out?

Comment (spco) [reply travis/run_tests.sh:211]: Sorry, just seen your edited comment about the tests. I think this is here to provide a summary of all the tests in one place at the end of the testsuite - so if something fails, you can see it here. I found it made the Travis logs a little easier to navigate. It's up to you whether you want to remove it or not.

**Comment by spco** [comment]
On the subject of the empty directories, I can't remember whether they are needed or not - do the tests complete if they're removed? Remember that git only tracks files, not directories, so an empty directory will not be replicate if you clone the repo.

If the directories do need to be kept for the tests to run etc (or just to make it clear to users what the directory structure should be - they can just look in any test directory), then the usual convention is to put a blank `.gitkeep` file inside them, so that git can keep track of them. They're not special files like `.gitignore`, so git doesn't treat them any differently to any other file.

File: travis/run_tests.sh (hunk 1 above)
Comment (rs028) [reply travis/run_tests.sh:211]: Yes I see that, but `make tests` does not execute the indent and style tests so this always report them as passed which is pointless. It makes sense only for `make alltests` in which case I agree it is useful. Unless I am reading this wrong.

**Comment by rs028** [comment]
The tests pass without the empty directories. The question is, will they remain in the repository since I cannot push their deletion? The rationale for removing them is to simplify the addition of new tests.

**Comment by spco** [comment]
Directories are not tracked if empty, so they will not show up - there is nothing to delete.

Good point about making it easier to add new tests. Is there a test that uses all of the directories? It might be useful to be able to point users to a single canonical example where they're all used, just to make it clear what is there. Or I guess the relevant section in the manual could be used as the guide instead anyway?

File: travis/run_tests.sh (hunk 1 above)
Comment (spco) [reply travis/run_tests.sh:211]: You're absolutely right.

Comment (rs028) [reply travis/run_tests.sh:211]: I will remove these lines then. But I agree with you that a summary should be printed at the end of the test suite. I will create a new issue about that.

**Comment by rs028** [comment]
Yes there are tests that use all directories. I think the manual is quite clear on how to add tests. In any case at some point we want to remake the whole test suite so that will be the moment to sort out these issues. I don't have time to work on the test suite now and I would like to have a new release fairly soon (btw I send you a bunch of messages on slack about that, don't know if you are checking it these days).

**Comment by rs028** [comment]
Why is this failing on macOS now?

**Comment by spco** [comment]
Have restarted it - the log is empty so it may just be a Travis issue.

**Comment by spco** [comment]
It's failed again in the same way. Nothing in their changelog since 9 days ago, so don't think it can be that. https://www.traviscistatus.com/ is showing no issues, but I think it's at their end. May have to wait a couple of hours and retry.

**Comment by rs028** [comment]
It was failing 2 days ago as well. It's not urgent, and at least it runs on Linux.

**Comment by rs028** [comment]
I think this is ready to go. I will wait a bit to merge while @mpanagi tests the solution to #413.

**Comment by mpanagi** [comment]
After the minor edits the model now compiles and runs and the results are the same.

**Comment by rs028** [comment]
Thanks @mpanagi.

If there is nothing else to add I will merge this and make a note somewhere about the travis tests for macOS.

**Comment by rs028** [comment]
Okay version 1.2 is now ready. Thanks to @spco, @bsn502, @mpanagi for your help and contributions!

**Pull request merged**